tools/GenerateNewControlProject.ps1

```diff
@@ -107,4 +107,67 @@ $xml.Save($testAppProject)
 FindAndReplaceInFile ($muxControlsDir + "\dev\Telemetry\RuntimeProfiler.h") "(\s*ProfId_Size.*\s*})" @"
 
         ProfId_$controlName,`$1
-"@
\ No newline at end of file
+"@
+# Randomize class name for multiple use of class in one powershell
+$id = get-random
+
+# We need double backslash for C# strings below
+$cleanMuxControlsDir = $muxControlsDir.Replace("\","\\") + "\\"
+
+echo "$cleanMuxControlsDir"
+
+$assemblies=(
+	"System","EnvDTE","EnvDTE80"
+)
+
+$source="
+using System;
+using EnvDTE;
+using EnvDTE80;
+using System.Collections.Generic;
+namespace SolutionHelper
+{
+	public static class SolutionRegister$id{
+		public static void Main(){
+            Console.WriteLine(`"Started script`");
+            var dteType = Type.GetTypeFromProgID(`"VisualStudio.DTE.16.0`", true);
+            var dte = (EnvDTE.DTE)System.Activator.CreateInstance(dteType);
```

Comment:
nit: might be worth adding an error message asking to install VS if we are not able to activate the type.

Reply:
Good idea, updated the script.